### PR TITLE
fix(gui-v2): handle page reset on cell update

### PR DESCRIPTION
**Pull request by pranavxc**
## Change Summary

- On cell update use response data to update row data rather than reloading the entire data.
- Audit row update 

re #3336

## Change type

- [ ] feat: (new feature for the user, not a new feature for build script)
- [x] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Provide summary of changes.

## Additional information / screenshots (optional)

Anything for maintainers to be made aware of

### CODE REVIEW

File: packages/nc-gui-v2/composables/useExpandedFormStore.ts [hunk 1]
```diff
@@ -14,6 +14,7 @@ import {
   useProvideSmartsheetRowStore,
 } from '#imports'
 import type { Row } from '~/composables'
+import { getHTMLEncodedText } from '~/utils'
```
Comment (bcakmakoglu) [inline packages/nc-gui-v2/composables/useExpandedFormStore.ts:17]: -> `'#imports'`

File: packages/nc-gui-v2/composables/useViewData.ts [hunk 2]
```diff
@@ -5,12 +5,12 @@ import { useNuxtApp } from '#app'
 import {
   IsPublicInj,
   NOCO,
-  ReloadViewDataHookInj,
   extractPkFromRow,
   extractSdkResponseErrorMsg,
   useProject,
   useUIPermission,
 } from '#imports'
+import { getHTMLEncodedText } from '~/utils'
```
Comment (bcakmakoglu) [inline packages/nc-gui-v2/composables/useViewData.ts:13]: -> `'#imports'`

File: packages/nc-gui-v2/composables/useViewData.ts [hunk 3]
```diff
@@ -166,43 +165,41 @@ export function useViewData(
     }
   }
 
-  const updateRowProperty = async (row: Record<string, any>, property: string) => {
+  const updateRowProperty = async (row: Row, property: string) => {
```
Comment (bcakmakoglu) [inline packages/nc-gui-v2/composables/useViewData.ts:168]: can we rename this argument to `toUpdate` or something? Using `toUpdate.row` would look cleaner than `row.row`

**Review by dstala — APPROVED** [review]

**Review by bcakmakoglu — CHANGES_REQUESTED** [review]

**Review by dstala — APPROVED** [review]

File: packages/nc-gui-v2/composables/useViewData.ts (hunk 3 above)
Comment (pranavxc) [reply packages/nc-gui-v2/composables/useViewData.ts:168]: sure :+1:

**Review by bcakmakoglu — APPROVED** [review]

**Pull request merged**